snapshots/output/syntax/src/infer-relationship.ts

```diff
@@ -76,60 +88,120 @@
+      property2: '3',
+//    ^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Strings#property2.
+    }
+    configuration2.property = configuration1.property
+//  ^^^^^^^^^^^^^^ reference local 14
+//                 ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+//                            ^^^^^^^^^^^^^^ reference local 11
+//                                           ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+    const configuration3: Configuration = {
+//        ^^^^^^^^^^^^^^ definition local 17
+//        documentation ```ts\nvar configuration3: Configuration\n```
+//                        ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Configuration#
+      property: 4,
+//    ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+      property2: '4',
+//    ^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Strings#property2.
+    }
+    return [
+      configuration1.property,
+//    ^^^^^^^^^^^^^^ reference local 11
+//                   ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+      configuration2.property,
+//    ^^^^^^^^^^^^^^ reference local 14
+//                   ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+      configuration3.property,
+//    ^^^^^^^^^^^^^^ reference local 17
+//                   ^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Numbers#property.
+    ]
+  }
+  
   export function returnStatementInsideArgumentExpression(): Configuration[] {
 //                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`infer-relationship.ts`/returnStatementInsideArgumentExpression().
 //                documentation ```ts\nfunction returnStatementInsideArgumentExpression(): Configuration[]\n```
 //                                                           ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Configuration#
     return [1].map<Configuration>(number => {
 //             ^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Array#map().
 //                 ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Configuration#
-//                                ^^^^^^ definition local 12
+//                                ^^^^^^ definition local 21
 //                                documentation ```ts\n(parameter) number: number\n```
       const incremented = number + 1
-//          ^^^^^^^^^^^ definition local 15
+//          ^^^^^^^^^^^ definition local 24
 //          documentation ```ts\nvar incremented: number\n```
-//                        ^^^^^^ reference local 12
+//                        ^^^^^^ reference local 21
       return {
         property: incremented,
-//      ^^^^^^^^ definition syntax 1.0.0 src/`infer-relationship.ts`/property7:
+//      ^^^^^^^^ definition syntax 1.0.0 src/`infer-relationship.ts`/property0:
```

Comment:
@SuperAuguste I don't know yet how to handle the case here but I think we can do it somehow by adding more entries to the `assignedType` map. Would you be interested in taking a look?

Reply:
Yep, definitely interested!